### PR TITLE
tests/int: cleanups

### PR DESCRIPTION
**Pull request by kolyshkin**
The ultimate idea is to parallelize running the integration tests.

So far just some preliminary cleanups, mostly resulting in simpler setup/cleanup
and laying the ground for parallel tests (every test is run in its own set of dirs and cgroups).

### CODE REVIEW

**Comment by kolyshkin** [comment]
OK I guess I reached the point where I need this one and https://github.com/opencontainers/runc/pull/2741 to move forward (to enable parallel bats tests execution).

File: tests/integration/helpers.bash
Comment (kolyshkin) [inline tests/integration/helpers.bash:28]: OK I guess this enabled checkpoint/restore tests on CentOS 7, and the lazy migration one failed with unclear diagnostics. Looking.

Comment (kolyshkin) [reply tests/integration/helpers.bash]: Moved this commit into https://github.com/opencontainers/runc/pull/2761 to "fix" the whole PR.

Comment (kolyshkin) [reply tests/integration/helpers.bash]: Moved the patch to a separate PR #2671.

**Comment by AkihiroSuda** [comment]
> BATS_RUN_TMPDIR (available since bats 1.2.1) is $BATS_TMPDIR/bats-run-$$
> (i.e. it is per bats run and includes PID), and it is cleaned up after
> the test (unless --no-tmpdir-cleanup is given).

Maybe we should error out if `$BATS_RUN_TMPDIR` is not set? (Does it already fail?)

**Review by AkihiroSuda — DISMISSED** [review]

**Comment by cyphar** [comment]
Needs rebase.

**Comment by kolyshkin** [comment]
Rebased, improved last commit description. I think this one is ready to go in.

**Review by AkihiroSuda — APPROVED** [review]

**Comment by kolyshkin** [comment]
@cyphar @mrunalp PTAL

**Review by mrunalp — APPROVED** [review]

**Comment by kolyshkin** [comment]
@opencontainers/runc-maintainers can someone please merge this one (I can't merge my own PRs)

**Comment by AkihiroSuda** [comment]
GitHub refusing clicking the merge button: `Required statuses must pass before merging` 

Probably needs rebase to retrigger Go 1.16.x tests.

**Comment by kolyshkin** [comment]
Wow, apparently I can merge my own PR if someone closes and reopens it :clown_face:

**Comment by AkihiroSuda** [comment]
Is it a vulnerability? 😄

**Pull request merged**
